lib/coverband/adapters/base.rb

```diff
@@ -64,24 +69,34 @@ def covered_files
 
       def split_coverage(types)
         types.reduce({}) do |data, type|
-          data.update(type => coverage(type))
+          if type == Coverband::RUNTIME_TYPE && Coverband.configuration.simulate_oneshot_lines_coverage
+            data.update(type => simulated_runtime_coverage)
+          else
+            data.update(type => coverage(type))
+          end
         end
       end
 
+      def simulated_runtime_coverage
```

Comment:
This is clever. I think one limitation is for code that's hit during eager_loading and runtime. Something we might want to point out to users.

Reply:
yeah, perhaps we should document that `simulated_runtime_coverage` has a nice advantage of overall reduced runtime memory usage, but it can't fully track eager vs runtime coverage and isn't as good as real oneshot for memory usage.